tests.py

```diff
@@ -1,129 +1,523 @@
-import contextlib
 import pathlib
 import platform
 import shutil
 import subprocess
 import sys
 import textwrap
+import re
+import threading
+import queue
+import time
+import email
+from http import client
 
 import pytest
 
 import sansio_lsp_client as lsp
 
 
-@contextlib.contextmanager
-def run_langserver(project_root, command):
+LOG_IN = False  # log data received from server to stdout
+LOG_OUT = False  # - sent - to stdout
+
+
+METHOD_DID_OPEN = "didOpen"
+METHOD_DID_CLOSE = "didClose"
+METHOD_DID_SAVE = "didSave"
+METHOD_DID_CHANGE = "didChange"
+
+METHOD_COMPLETION = "completion"
+METHOD_HOVER = "hover"
+METHOD_SIG_HELP = "signatureHelp"
+METHOD_DEFINITION = "definition"
+METHOD_REFERENCES = "references"
+METHOD_IMPLEMENTATION = "implementation"
+METHOD_DECLARATION = "declaration"
+METHOD_TYPEDEF = "typeDefinition"
+METHOD_DOC_SYMBOLS = "documentSymbol"
+METHOD_FORMAT_DOC = "formatting"
+METHOD_FORMAT_SEL = "rangeFormatting"
```

Comment:
I would prefer just writing the strings in the code. This is not an Enterprise Java project with a strict policy.

Reply:
That would be error-prone and more difficult to navigate.

Reply:
I'm ok with them written out as strings like this. Not something I'd've personally done myself, but it's understandable. .